paper/src/main/java/vg/civcraft/mc/citadel/listener/BlockListener.java

```diff
@@ -547,4 +549,20 @@ public void onMossSpread(BlockFertilizeEvent event) {
 			}
 		}
 	}
+
+	@EventHandler(ignoreCancelled = true)
+	public void onDispenserGrowMoss(BlockDispenseEvent event){
```

Comment:
Is this something for Citadel? I would assume something like this would be in SAH, unless you plan to add reinforcement checks to it here.

Reply:
This is purely to prevent moss spread occuring which would take over neighbouring stone/dirt and could be reinforced

Reply:
Would it be reasonable to check the reinforcements, and cancel if the stone is reinforced, but not on the same group as the dispenser or original stone?

This also doesn't prevent a player from spreading by hand it seems.

Reply:
Paper has no method that provides the blocks affected by moss spread when done via dispenser, hence this approach while being the least favourable, also being one of the only

Player bonemealing is handled above since that event provides the blocks modified so we can prevent it better in that scenario

Reply:
As per https://github.com/CivMC/Citadel/blob/eea1563e0668768f8cf1b4eeaa1ea9996a0facff/paper/src/main/java/vg/civcraft/mc/citadel/listener/BlockListener.java#L543

Reply:
10-4, I opened #12 to track this for the future if another solution ever becomes available